packages/amplify-cli/src/init-steps/s0-analyzeProject.ts

```diff
@@ -242,7 +242,7 @@ async function getEnvName(context) {
   } else {
     const allEnvs = context.amplify.getAllEnvs();
 
-    if (allEnvs.length > 0) {
+    if (allEnvs.length > 0 && addNewEnv(context).exec === false) {
```

Comment:
Instead of calling `addNewEnv()` repeatedly, please store the result in a variable.

Reply:
Resolved!